gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/common/impl/InMemoryTopology.java

```diff
@@ -267,7 +269,8 @@ private static String getCircularDependencyChain(ConfigKeyPath initialConfigKey,
 
   private List<ConfigKeyPath> getOwnImportsFromCache(ConfigKeyPath configKey) {
     if (this.ownImportMap.containsKey(configKey)) {
-      return this.ownImportMap.get(configKey);
+      List<ConfigKeyPath> configKeyPaths = new ArrayList<>(this.ownImportMap.get(configKey));
```

Comment:
The error is raised from the cache. How does this fix that? Also, if concurrent modification is during traversal of the list, the copy can also hit the same exception.
```com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4721)
21-06-2017 06:25:07 PDT tracking-hourly INFO - 	at gobblin.config.common.impl.InMemoryValueInspector.getResolvedConfig(InMemoryValueInspector.java:104)
21-06-2017 06:25:07 PDT tracking-hourly INFO - 	at```